ingest/common/pooltracker/memory_pool_tracker.go

```diff
@@ -0,0 +1,93 @@
+package pooltracker
+
+import (
+	"github.com/osmosis-labs/osmosis/v25/ingest/sqs/domain"
+	poolmanagertypes "github.com/osmosis-labs/osmosis/v25/x/poolmanager/types"
+)
+
+// poolBlockUpdateTracker is a struct that tracks the pools that were updated in a block.
+type poolBlockUpdateTracker struct {
+	concentratedPools             map[uint64]poolmanagertypes.PoolI
+	concentratedPoolIDTickChange  map[uint64]struct{}
+	cfmmPools                     map[uint64]poolmanagertypes.PoolI
+	cosmwasmPools                 map[uint64]poolmanagertypes.PoolI
+	cosmwasmPoolsAddressToPoolMap map[string]poolmanagertypes.PoolI
+}
+
+// NewMemory creates a new memory pool tracker.
+func NewMemory() domain.BlockPoolUpdateTracker {
```

Comment:
Note to reviewer: this abstraction and tests are simply moved from `sqs_pool_tracker.go` and renamed